src/Workspaces/CoreTest/Execution/SnapshotSerializationTests.cs

```diff
@@ -732,24 +732,32 @@ private async Task<Solution> GetSolutionAsync(IRemotableDataService service, Pin
                     continue;
                 }
 
-                var documents = new List<DocumentInfo>();
-                foreach (var documentObject in projectObject.Documents.ToDocumentObjects(service))
+                async Task<List<DocumentInfo>> CreateDocumentInfosAsync(ChecksumObjectCollection<DocumentStateChecksums> checksums)
                 {
-                    var documentInfo = await service.GetValueAsync<DocumentInfo.DocumentAttributes>(documentObject.Info).ConfigureAwait(false);
-                    var text = await service.GetValueAsync<SourceText>(documentObject.Text).ConfigureAwait(false);
-
-                    // TODO: do we need version?
-                    documents.Add(
-                        DocumentInfo.Create(
-                            documentInfo.Id,
-                            documentInfo.Name,
-                            documentInfo.Folders,
-                            documentInfo.SourceCodeKind,
-                            TextLoader.From(TextAndVersion.Create(text, VersionStamp.Create())),
-                            documentInfo.FilePath,
-                            documentInfo.IsGenerated));
+                    List<DocumentInfo> infos = new List<DocumentInfo>();
+
+                    foreach (var documentStateChecksums in checksums)
+                    {
+                        var documentInfo = await service.GetValueAsync<DocumentInfo.DocumentAttributes>(documentStateChecksums.Info).ConfigureAwait(false);
```

Comment:
Chatting with @heejaechang we noticed that this is more or less duplicating what is in SolutionCreator.cs. We've filed #35335 to track removing the duplication. There's some architectural stuff in the way -- some interfaces would need to be redefined to make it all work.